### PR TITLE
[Fix-14704][dao] pg not support backquote

### LINKED ISSUE #14704: [Bug] [dao] select DQ rule page failed
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

image

When using pg, backquote identifiers are not supported. So an error was reported when querying the data quality rule pagination interface.

### What you expected to happen

refer What happened

### How to reproduce

refer What happened

### Anything else

_No response_

### Version

dev

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by eye-gu**
## Purpose of the pull request

close #14704

## Brief change log

rm backquote in sql

### CODE REVIEW

File: dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/DqRuleInputEntryMapper.xml [hunk 1]
```diff
@@ -40,4 +40,26 @@
         FROM t_ds_relation_rule_input_entry where rule_id = #{ruleId} )  b
         on a.id = b.rule_input_entry_id order by b.index
     </select>
+
+    <select id="getRuleInputEntryList" databaseId="pg" resultType="org.apache.dolphinscheduler.dao.entity.DqRuleInputEntry">
```
Comment (caishunfeng) [inline dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/DqRuleInputEntryMapper.xml:44]: Hi @eye-gu Although this is also a solution, but it means that we have to maintain two same sql.
Can this sql be splited into simple query? WDYT?

Comment (eye-gu) [reply dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/DqRuleInputEntryMapper.xml:44]: Removing the backquote will cause unit testing to fail.

Comment (eye-gu) [reply dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/DqRuleInputEntryMapper.xml:44]: Alternatively refer to Shenyu, which uses an Interceptor to replace backquotes with double quotes.
https://github.com/apache/shenyu/blob/master/shenyu-admin/src/main/java/org/apache/shenyu/admin/mybatis/pg/interceptor/PostgreSQLPrepareInterceptor.java

Comment (caishunfeng) [reply dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/DqRuleInputEntryMapper.xml:44]: WDYT? @zhongjiajie @ruanwenjun

Comment (qingwli) [reply dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/DqRuleInputEntryMapper.xml:44]: And there is another solution is to change this column name and totally avoid questions like this

Comment (qingwli) [reply dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/DqRuleInputEntryMapper.xml:44]: https://github.com/apache/dolphinscheduler/pull/14653 here is another solution

Comment (eye-gu) [reply dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/DqRuleInputEntryMapper.xml:44]: I  think the #14653 solution is better

**Pull request closed without merging**